examples/advanced/geane/src/FairGeaneTr.cxx

```diff
@@ -0,0 +1,148 @@
+/********************************************************************************
+ *    Copyright (C) 2014 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+/**
+ * FairGeaneTr.cxx
+ *
+ * @since 2019-09-10
+ * @author R. Karabowicz
+ */
+
+#include "FairGeaneTr.h"
+
+#include "TClonesArray.h"
+#include "TGeant3TGeo.h"
+#include "TGeant3.h"
+#include "TVector3.h"
+#include "TTree.h"
+#include "TDatabasePDG.h"
+
+#include "FairRootManager.h"
+#include "FairLogger.h"
+#include "FairTrackParP.h"
+#include "PixelPoint.h"
+
+// -----   Default constructor   -------------------------------------------
+FairGeaneTr::FairGeaneTr()
+    : FairTask("FairGeanePropagator")
+    , fPro(0)
+{ }
+// -------------------------------------------------------------------------
+
+// -----   Destructor   ----------------------------------------------------
+FairGeaneTr::~FairGeaneTr() { }
+// -------------------------------------------------------------------------
+
+// -----   Public method Init   --------------------------------------------
+InitStatus FairGeaneTr::Init() {
+
+     // Get RootManager
+    FairRootManager* ioman = FairRootManager::Instance();
+    if ( ! ioman ) {
+        LOG(error) << "FairGeaneTr::Init: RootManager not instantised!";
+        return kFATAL;
+    }
+
+    fTrackParIni = new TClonesArray("FairTrackParP");
```

Comment:
Who is the owner of the `fTrackPar*` members? Unless I am missing another ROOT magic all those `TClonesArray`s are never deleted? If yes, then let's make them non-pointer members.